src/services/checkSavedApplicationService.ts

```diff
@@ -1,25 +1,32 @@
 import { Response } from "express";
 import Resource from "@companieshouse/api-sdk-node/dist/services/resource";
-import { TransactionList } from "@companieshouse/api-sdk-node/dist/services/transaction/types";
+import { TransactionData, TransactionList } from "@companieshouse/api-sdk-node/dist/services/transaction/types";
 import { Session } from "@companieshouse/node-session-handler";
-import { APPROVED, CLOSED, REJECTED } from "../common/__utils/constants";
+import { APPROVED, CLOSED, REJECTED, RESUME_APPLICATION_ID } from "../common/__utils/constants";
 import { BASE_URL, CANNOT_REGISTER_AGAIN, CANNOT_SUBMIT_ANOTHER_APPLICATION, CHECK_SAVED_APPLICATION, SAVED_APPLICATION, TYPE_OF_BUSINESS } from "../types/pageURL";
 import logger from "../utils/logger";
 import { deleteAcspApplication } from "./acspRegistrationService";
 import { ErrorService } from "./errorService";
 
 export const getRedirectionUrl = async (transactionlistResource: Resource<TransactionList>, session: Session, res: Response, locales:any, lang:string): Promise<string> => {
     const transactionList = transactionlistResource.resource;
-    const transaction = transactionList?.items[transactionList?.items.length - 1];
+    const length = transactionList!.items.length;
+    for (let i = 0; i < length; i++) {
+        logger.debug("element------>" + transactionList!.items[i].id);
+    }
+    const transaction: TransactionData = transactionList!.items[0];
     logger.debug("transactionId: " + transaction?.id);
+    const applicationId = getApplicationId(transaction);
+
     var url = "";
     if (transaction?.status !== CLOSED) {
```

Comment:
Should we check for length of the list as well. If length is 0 then the redirection url would be type of business page right?

Reply:
When there are no transactions, we get 404 status code and the redirection url is type of business. This is handled in checkSavedApplicationController
The compiler reaches above code only when we get 200 status code from transaction api. So we don't need to check for length is 0.

Reply:
Gotcha